streams/src/test/java/org/apache/kafka/streams/state/internals/SegmentsTest.java

```diff
@@ -241,28 +244,34 @@ public void futureEventsShouldNotCauseSegmentRoll() {
     }
 
     private void updateStreamTimeAndCreateSegment(final int segment) {
-        context.setStreamTime(segmentInterval * segment);
+        context.setStreamTime(SEGMENT_INTERVAL * segment);
         segments.getOrCreateSegmentIfLive(segment, context);
     }
 
     @Test
     public void shouldUpdateSegmentFileNameFromOldDateFormatToNewFormat() throws Exception {
+        final long segmentInterval = 60_000L; // the old segment file's naming system maxes out at 1 minute granularity.
+
+        segments = new Segments(storeName, NUM_SEGMENTS * segmentInterval, segmentInterval);
+
         final String storeDirectoryPath = stateDirectory.getAbsolutePath() + File.separator + storeName;
         final File storeDirectory = new File(storeDirectoryPath);
+        //noinspection ResultOfMethodCallIgnored
         storeDirectory.mkdirs();
 
         final SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMddHHmm");
         formatter.setTimeZone(new SimpleTimeZone(0, "UTC"));
 
         for (int segmentId = 0; segmentId < NUM_SEGMENTS; ++segmentId) {
             final File oldSegment = new File(storeDirectoryPath + File.separator + storeName + "-" + formatter.format(new Date(segmentId * segmentInterval)));
+            //noinspection ResultOfMethodCallIgnored
             oldSegment.createNewFile();
         }
 
         segments.openExisting(context);
 
         for (int segmentId = 0; segmentId < NUM_SEGMENTS; ++segmentId) {
-            final File newSegment = new File(storeDirectoryPath + File.separator + storeName + "." + segmentId * (retentionPeriod / (NUM_SEGMENTS - 1)));
+            final File newSegment = new File(storeDirectoryPath + File.separator + segments.segmentName(segmentId));
```

Comment:
We should assemble the new format manually -- otherwise, if `segmentName()` is change accidentally, we would not notice.